src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -462,14 +653,13 @@ private void retrieveStreamStats(Map<String, String> stats) {
 
 				if (streamData != null) {
 					List<Stream> streams = streamData.getStreams();
-
 					for (Stream stream : streams) {
 						StreamInfo streamInfo = stream.getStreamInfo();
 						StreamStats streamStats = stream.getStreamStats();
 
 						// Stream name filtering
 						if (this.streamName != null && streamNameSet.contains(streamInfo.getName())) {
```

Comment:
one of the valid values that handleAdapterPropertiesInputFromUser populates is null, which would cause npe (also for streamStatusSet and portNumberSet)

Reply:
I fixed it